### PR TITLE
[OPS-1448] Replace buildkite with github actions

### PR DESCRIPTION
**Pull request by Sereja313**
## Description
Problem: The current buildkite-based ci is being replaced with github actions.

Solution: Add github ci config, remove buildkite config.

## Related issue(s)

Fixed #

#### Related changes (conditional)

- Tests
  - [ ] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)
    - Haddock

#### Stylistic guide (mandatory)

- [x] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).
- [x] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+# SPDX-License-Identifier: MPL-2.0
+
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+      - staging
```
Comment (dcastro) [inline .github/workflows/check.yml:12]: Can we just trigger the CI on every push to any branch?
Sometimes it's useful to see if the CI is still green, while working on a branch, even before opening a PR.

```suggestion
  push:
```

File: .github/workflows/check.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    runs-on: [self-hosted, nix]
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: check trailing whitespaces
+        run: nix build -L .#checks.x86_64-linux.trailing-whitespace
+
+      - name: shellcheck
+        run: nix build -L .#checks.x86_64-linux.shellcheck
+        if: success() || failure()
```
[the remark this answers is not in the document]
Comment (Sereja313) [reply .github/workflows/check.yml:25]: do you mean `if: success() || failure()`?  It allows a step to be executed even if the previous step failed.

File: .github/workflows/check.yml (hunk 1 above)
Comment (Sereja313) [reply .github/workflows/check.yml:12]: Sure

File: .github/workflows/check.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (dcastro) [reply .github/workflows/check.yml:25]: Ah gotcha

**Review by dcastro — APPROVED** [review]

**Pull request merged**